### ISSUE TITLE
feat: add consistent-list-marker rule (MD004)

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#202]
## Overview

Implement the `consistent-list-marker` rule (equivalent to markdownlint MD004 / remark-lint `unordered-list-marker-style`).

Closes part of #76.

## Problem

Markdown allows three characters as unordered list markers: `-`, `*`, and `+`. All three are valid, so a document can mix them freely — but doing so makes the source harder to read and edit consistently. A reader scanning the raw Markdown has to mentally parse each marker variant, and a contributor editing the file may accidentally introduce a different marker than the rest of the document uses.

```markdown
- First item
* Second item
+ Third item
```

## Proposed Rule Key

`consistent-list-marker`

## Detection Logic

Scan each line (outside fenced code blocks) for unordered list item openers. A line is a list item opener if it matches `/^(\s*)([-*+])\s+\S/` — that is, optional leading whitespace, one of `-`, `*`, or `+`, followed by a space and a non-whitespace character.

Depending on the configured `style`:

| Style | Behaviour |
|---|---|
| `consistent` (default) | The **first** marker found in the document sets the expected style; any subsequent list item using a different marker is flagged |
| `dash` | Every list item must use `-`; other markers are always flagged |
| `asterisk` | Every list item must use `*`; other markers are always flagged |
| `plus` | Every list item must use `+`; other markers are always flagged |

**Invalid** (style: `consistent`, first marker is dash):
```markdown
- First item
* Second item
```

**Valid** (style: `consistent`):
```markdown
- First item
- Second item
- Third item
```

**Valid** (style: `dash`):
```markdown
- Item one
- Item two
```

**Notes:**
- Ordered list items (`1.`, `2.`, etc.) are not affected.
- Content inside fenced code blocks is ignored.

## Fix

Replace the flagged list marker with the expected one. For example, if the expected style is dash, change `* item` to `- item`.

## Implementation Notes

- Rule key: `consistent-list-marker`
- Config field: `style` string (`"consistent"` | `"dash"` | `"asterisk"` | `"plus"`), default `"consistent"`
- Default enabled with `style: "consistent"`
- Skip content inside fenced code blocks (reuse `openingFenceMarker`, `IsClosingFence`)
- No inline code stripping needed (markers are at the start of lines)
- Use `firstNonSpaceByte` to prefilter lines

### Config

**Default (`DefaultConfigJSON`):**
```json
"consistent-list-marker": { "style": "consistent" }
```

**Default (`Default()` in Go):**
```go
"consistent-list-marker": {
    Enabled:  true,
    Severity: SeverityError,
    Options:  map[string]interface{}{"style": "consistent"},
},
```

**User customisation examples:**
```json
"consistent-list-marker": { "style": "dash" }
```
```json
"consistent-list-marker": false
```

## References

- [markdownlint MD004](https://github.com/DavidAnson/markdownlint/blob/main/doc/md004.md)
- [remark-lint unordered-list-marker-style](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-unordered-list-marker-style)

**Comment by shinagawa-web** [comment]
## Implementation Plan

### Files to create

- \`internal/rule/consistent_list_marker.go\` — rule implementation
- \`internal/rule/consistent_list_marker_test.go\` — unit tests
- \`e2e/config-consistent-list-marker.json\` — E2E config
- \`e2e/fixtures/consistent_list_marker_valid.md\` — valid fixture
- \`e2e/fixtures/consistent_list_marker_violation.md\` — violation fixture

### Files to modify

- \`internal/config/config.go\` — add to \`DefaultConfigJSON\` and \`Default()\`
- \`internal/linter/linter.go\` — add \`consistentListMarkerStyle()\` helper and wire into \`collectLineErrors()\`
- \`internal/linter/linter_test.go\` — add integration tests for 100% coverage
- \`e2e/e2e_test.go\` — add \`ConsistentListMarkerValid\`, \`ConsistentListMarkerViolation\` cases; update \`DirectoryRecursion\` file count
- \`cmd/root_bench_test.go\` — add mixed-marker-free list content to \`writeList()\`
- \`docs/content/docs/rules.md\` — add row for \`consistent-list-marker\`
- \`docs/content/docs/roadmap.md\` — mark as completed

### Checklist

- [ ] Implement \`CheckConsistentListMarker\` in \`internal/rule/consistent_list_marker.go\`
- [ ] Unit tests covering: all 4 styles, mixed markers, nested lists, fenced code block skip, offset, invalid style fallback
- [ ] Register in \`config.go\` (both \`DefaultConfigJSON\` and \`Default()\`)
- [ ] Wire into \`linter.go\` (\`consistentListMarkerStyle()\` helper + \`collectLineErrors\` call)
- [ ] Linter integration tests for 100% coverage of new helper
- [ ] E2E fixtures and test cases
- [ ] \`TestBenchmarkContentIsViolationFree\` passes
- [ ] \`make test-all\` passes
- [ ] \`make static-lint\` passes (gofmt, gocyclo ≤15, gocognit ≤30)
